account/onboarding.go

```diff
@@ -0,0 +1,125 @@
+package account
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/pborman/uuid"
+	"github.com/status-im/status-go/extkeys"
+)
+
+var errInvalidMnemonicPhraseLength = errors.New("invalid mnemonic phrase length")
+
+// OnboardingAccount is returned during onboarding and contains its ID and the mnemonic to re-generate the same account Info keys.
+type OnboardingAccount struct {
+	ID       string `json:"id"`
+	mnemonic string
+	Info     Info `json:"info"`
+}
+
+// Onboarding is a struct contains a slice of OnboardingAccount.
+type Onboarding struct {
+	accounts map[string]*OnboardingAccount
+}
+
+// NewOnboarding returns a new onboarding struct generating n accounts.
+func NewOnboarding(n, mnemonicPhraseLength int) (*Onboarding, error) {
+	onboarding := &Onboarding{
+		accounts: make(map[string]*OnboardingAccount),
+	}
+
+	for i := 0; i < n; i++ {
+		account, err := onboarding.generateAccount(mnemonicPhraseLength)
+		if err != nil {
+			return nil, err
+		}
+		uuid := uuid.NewRandom().String()
+		account.ID = uuid
+		onboarding.accounts[uuid] = account
+	}
+
+	return onboarding, nil
+}
+
+// Accounts return the list of OnboardingAccount generated.
+func (o *Onboarding) Accounts() []*OnboardingAccount {
+	accounts := make([]*OnboardingAccount, 0)
```

Comment:
```suggestion
	accounts := make([]*OnboardingAccount, len(o.accounts))
```

Reply:
@adambabik I put it back as it was. I forgot I was transforming from hash to slice, so there wasn't a clean way to get the index.